tests/rptest/tests/rpk_acl_test.py

```diff
@@ -36,6 +36,37 @@ def __init__(self, test_ctx, *args, **kwargs):
                             sasl_mechanism=self.mechanism)
         self.superuser = self.redpanda.SUPERUSER_CREDENTIALS
 
+    def _superclient(self):
+        return RpkTool(self.redpanda,
+                       username=self.superuser.username,
+                       password=self.superuser.password,
+                       sasl_mechanism=self.mechanism)
+
+    @cluster(num_nodes=3)
+    def test_modify_then_query(self):
+        """
+        This test ensures that sending a command to enable an ACL then
+        querying it (on a node that may be a follower) returns the most
+        up to data result
+        """
+        # All operations apart from 'ALL'
+        operations = [
+            'DESCRIBE', 'READ', 'WRITE', 'CREATE', 'DELETE', 'ALTER',
+            'DESCRIBE_CONFIGS', 'ALTER_CONFIGS'
+        ]
+        superclient = self._superclient()
+        superclient.acl_create_allow_cluster(self.superuser.username,
+                                             "describe")
+
+        for op in operations:
+            self._rpk.sasl_allow_principal(f'User:{self.username}', [op],
+                                           'topic', 'foo',
+                                           self.superuser.username,
+                                           self.superuser.password,
+                                           self.superuser.algorithm)
+            described = superclient.acl_list()
```

Comment:
q: does it always go to a leader? in which case it always returns the freshest response, wondering if we can loop through all the brokers.

Reply:
> [bharathv](https://github.com/bharathv) [3 weeks ago](https://github.com/redpanda-data/redpanda/pull/12716/commits/be022545b06b3094eb03e602d69340712f8c0168#r1303314308)
q: does it always go to a leader? in which case it always returns the freshest response, wondering if we can loop through all the brokers.

@graphcareful ?

Reply:
I spoke about this to Bharath offline, probably should have updated the comments here, in any case the answer is a random broker will be chosen